src/Avalonia.Animation/Animation.cs

```diff
@@ -153,5 +160,28 @@ public IDisposable Apply(Animatable control, IObservable<bool> matchObs)
             }
             return this;
         }
+
+        /// <inheritdocs/>
+        public Task Run(Animatable control)
+        {
+            var tcs = new TaskCompletionSource<object>();
+
+            if (this.RepeatCount == RepeatCount.Loop)
+                tcs.SetException(new InvalidOperationException("Looping animations must not use the Run method."));
+
+            this.Done += delegate
```

Comment:
We should check that the sender parameter is equal to control so we can run the same animation on multiple controls programmatically.

Reply:
Good catch!